src/urllib3/util/url.py

```diff
@@ -322,17 +322,14 @@ def _idna_encode(name):
 
 def _encode_target(target):
     """Percent-encodes a request target so that there are no invalid characters"""
-    if not target.startswith("/"):
-        return target
-
     path, query, fragment = TARGET_RE.match(target).groups()
     target = _encode_invalid_chars(path, PATH_CHARS)
     query = _encode_invalid_chars(query, QUERY_CHARS)
     fragment = _encode_invalid_chars(fragment, FRAGMENT_CHARS)
     if query is not None:
         target += "?" + query
     if fragment is not None:
-        target += "#" + target
+        target += "#" + fragment
```

Comment:
Nice bug fix!

Reply:
Yeah, only 7 uncovered lines, and one bug! I should have seen it when reviewing the PR, but didn't :)

Reply:
I'm pretty sure we shouldn't be sending this in our request tbh. Request target is path+query.

Reply:
Resolving as this was fixed in #1732.

Reply:
Resolving as this was fixed in #1732.